src/TSTransformer/util/assertNever.ts

```diff
@@ -0,0 +1,51 @@
+import { spawnSync } from "child_process";
+import kleur from "kleur";
+import { LogService } from "Shared/classes/LogService";
+import { getKindName } from "TSTransformer/util/getKindName";
+import ts from "typescript";
+import util from "util";
+
+type LsInfo = {
+	name: string;
+	version: string;
+	dependencies: Record<string, LsInfo>;
+};
+
+function findTypescriptVersion(info: LsInfo): string | undefined {
+	if (info.name === "roblox-ts" && info.dependencies.typescript) {
+		return info.dependencies.typescript.version;
+	}
+	for (const [, dep] of Object.entries(info.dependencies)) {
+		const found = findTypescriptVersion(dep);
+		if (found) {
+			return found;
+		}
+	}
+}
+
+function error(message: string): never {
+	/* istanbul ignore */
+	const typescriptVersion = findTypescriptVersion(
+		JSON.parse(spawnSync("npm ls typescript --json").stdout.toString()) as LsInfo,
+	);
+	LogService.fatal(
+		kleur.red(`Exhaustive assertion failed! ${message}`) +
+			kleur.yellow("\nThis is usually caused by a TypeScript version mismatch.") +
+			kleur.yellow("\nMake sure that all TS versions in your project are the same.") +
+			kleur.yellow("\nYou can check the list of installed versions with `npm list typescript`") +
+			(typescriptVersion ? kleur.yellow(`\nTry running \`npm install typescript@=${typescriptVersion}\``) : ""),
+	);
+}
+
+/**
+ * Asserts at compile-time that `value` is `never`, throws at runtime.
+ * @param value The value to check the exhaustiveness of
+ * @param message The message of the error
+ */
+export function assertNever(value: never, message: string): never {
+	/* istanbul ignore */
+	const isTsNode = "kind" in value && ts.isNode(value);
```

Comment:
```suggestion
	const isTsNode = typeof value === "object" && "kind" in value && ts.isNode(value);
```

If `value` is `undefined`, this would error. We can't assume the type of `value` since the type is `never`.